execution_engine/src/storage/transaction_source/lmdb.rs

```diff
@@ -131,7 +203,8 @@ impl<'a> TransactionSource<'a> for LmdbEnvironment {
         self.env.begin_ro_txn()
     }
 
-    fn create_read_write_txn(&'a self) -> Result<RwTransaction<'a>, Self::Error> {
+    fn create_read_write_txn(&'a self) -> Result<Self::ReadWriteTransaction, Self::Error> {
+        self.resize_if_required()?;
```

Comment:
I don't think we should be resizing on every transaction creation. Moreover, it would check the DB size many times per second in our case and it's not needed. The buffer (and if necessary the resize) is big enough to do it less frequently.